### DISCUSSION TITLE
Directory stack locking

### DISCUSSION THREAD

**Discussion by typecasto** [fish-shell/fish-shell#8946] [Ideas]
Often times I will have a script that uses `pushd` and `popd` to manage the directory stack, but sometimes something fails and it ends up `popd`ing too many times. If this happens, I'd like to have it not mess with the directory stack as it was before the function executed. Thus, my idea is `pushd --lock` and `--unlock`. Whenever you lock the stack, entries that were in it at the time of lock are unusable, meaning if you try to run `pushd ~/whatever; pushd --lock; popd` it will give you an error. That way, running popd too many times has no effect, instead of clearing the directory stack and continuing function execution in an unintended directory.

**Comment by faho** [comment]
I'm struggling to come up with a case where

1. pushd can fail
2. I'd want to still go on

So what I would do is

    pushd ~/whatever
    or return

or something like

    if pushd ~/whatever
       # do your thing
       popd
    end

>continuing function execution in an unintended directory

Once pushd failed you're *already* executing in an unintended directory. Not doing popd is the least of your problems.
